app/assets/javascripts/user.js

```diff
@@ -223,4 +223,62 @@ $(document).ready(function () {
   $("#read_tou").on("click", function () {
     $("#continue").prop("disabled", !($(this).prop("checked") && $("#read_ct").prop("checked")));
   });
+
+
+  const isTaken /* Record<string, string> */ = {}
+
+  function indicateInvalidDisplayName(isValid /* boolean | undefined*/) {
+    // isValid might also be 'undefined', hence we check explicitly for 'true' and 'false'
+    $("#user_display_name").toggleClass("is-valid", isValid === true)
+    $("#user_display_name").toggleClass("is-invalid", isValid === false)
+  }
+
+  function checkDisplayName(){
+    const displayName = $("#user_display_name").val()
+    if(displayName === ""){
+      indicateInvalidDisplayName(false)
+      return;
+    }
+    if(displayName.match(/[\/;.,?%#]/)!== null) {
+      // We detected an invalid character
+      indicateInvalidDisplayName(false)
+      return
+    }
+    indicateInvalidDisplayName(!isTaken[displayName])
```

Comment:
`!undefined` is going to be true, the username will be indicated as valid until the fetch completes